### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-tf-module.git`](https://github.com/cisagov/skeleton-tf-module.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/openvpn-server-tf-module.git openvpn-server-tf-module
    cd openvpn-server-tf-module
    git remote add skeleton https://github.com/cisagov/skeleton-tf-module.git
    git remote set-url --push skeleton no_push
    git switch develop
    git checkout -b lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the
   branch, `commit`, and `push` your changes:

    ```console
    git add .github/CODEOWNERS README.md examples/basic_usage/README.md examples/basic_usage/variables.tf main.tf outputs.tf variables.tf versions.tf 
    git commit
    git push --force --set-upstream origin lineage/skeleton
    ```

    Note that you may _append_ to the default merge commit message
    that git creates for you, but *please do not delete the existing
    content*.  It provides useful information about the merge that is
    being performed.

1. Wait for all the automated tests to pass.

1. Check the "Everything is cool" checkbox below:

    - [x] ✌️ The conflicts in this pull request have been resolved.

1. Mark this draft pull request "Ready for review".

------------

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

File: examples/basic_usage/main.tf [hunk 1]
```diff
@@ -1,37 +1,53 @@
...
-  region = var.aws_region # route53 is global, but still required by terraform
+  alias = "dns"
   assume_role {
     role_arn     = var.dns_role_arn
     session_name = "terraform-openvpn-dns"
   }
+  default_tags {
+    tags = local.tags
+  }
+  region = var.aws_region # route53 is global, but still required by terraform
```
Comment (dav3r) [inline examples/basic_usage/main.tf]:
```suggestion
  region = var.aws_region # Route 53 is global, but still required by Terraform
```

**Review by dav3r — APPROVED** [review]
👍  (with a minor capitalization change requested)

File: examples/basic_usage/README.md [hunk 2]
```diff
@@ -56,9 +56,9 @@ No requirements.
...
 |------|-------------|
-| arn | The EC2 instance ARN |
-| availability\_zone | The AZ where the EC2 instance is deployed |
-| id | The EC2 instance ID |
-| private\_ip | The private IP of the EC2 instance |
+| arn | The EC2 instance ARN. |
+| availability\_zone | The AZ where the EC2 instance is deployed. |
+| id | The EC2 instance ID. |
+| private\_ip | The private IP of the EC2 instance. |
 | public\_ip | The public IP of the EC2 instance |
```
Comment (mcdonnnj) [inline examples/basic_usage/README.md]: Output description would need to be changed.

```suggestion
| public\_ip | The public IP of the EC2 instance. |
```

File: README.md [hunk 3]
```diff
@@ -130,7 +129,6 @@ module "example" {
 | ssm\_region | The region of the SSM to access. | `string` | `"us-east-1"` | no |
 | ssm\_tlscrypt\_key | The SSM key that contains the tls-auth key. | `string` | `"/openvpn/server/tlscrypt.key"` | no |
 | subnet\_id | The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0) | `string` | n/a | yes |
```
Comment (mcdonnnj) [inline README.md]: Variable description would need to be updated.

```suggestion
| subnet\_id | The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0). | `string` | n/a | yes |
```

File: README.md [hunk 4]
```diff
@@ -130,7 +129,6 @@ module "example" {
...
-| tags | Tags to apply to all AWS resources created | `map(string)` | `{}` | no |
 | trusted\_cidr\_blocks\_vpn | A list of the CIDR blocks that are allowed to access the VPN port on OpenVPN servers (e.g. ["10.10.0.0/16", "10.11.0.0/16"]) | `list(string)` | n/a | yes |
```
Comment (mcdonnnj) [inline README.md]: Variable description would need to be updated.

```suggestion
| trusted\_cidr\_blocks\_vpn | A list of the CIDR blocks that are allowed to access the VPN port on OpenVPN servers (e.g. ["10.10.0.0/16", "10.11.0.0/16"]). | `list(string)` | n/a | yes |
```

File: README.md [hunk 5]
```diff
@@ -130,7 +129,6 @@ module "example" {
...
 | ttl | The TTL value to use for Route53 DNS records (e.g. 86400).  A smaller value may be useful when the DNS records are changing often, for example when testing. | `number` | `60` | no |
 | vpn\_group | The LDAP group that grants users the permission to connect to the VPN server. (e.g. vpnusers) | `string` | n/a | yes |
```
Comment (mcdonnnj) [inline README.md]: Variable description would need to be updated.

```suggestion
| vpn\_group | The LDAP group that grants users the permission to connect to the VPN server (e.g. vpnusers). | `string` | n/a | yes |
```

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔ I also had some description suggestions.

File: examples/basic_usage/README.md (hunk 2 above)
Comment (jsf9k) [reply examples/basic_usage/README.md]: See commit f0aecc6.

File: README.md (hunk 3 above)
Comment (jsf9k) [reply README.md]: See commit f0aecc6.

File: README.md (hunk 4 above)
Comment (jsf9k) [reply README.md]: See commit f0aecc6.

File: README.md (hunk 5 above)
Comment (jsf9k) [reply README.md]: See commit f0aecc6.

**Review by mcdonnnj — APPROVED** [review]
I don't want to derail this lineage update so I'm approving because you've fixed my requests. However, I noticed that there are three variables with defaults in the `Required parameters` section:
https://github.com/cisagov/openvpn-server-tf-module/blob/f0aecc6996effc288324204086453ee2e4671d33/variables.tf#L12-L16
https://github.com/cisagov/openvpn-server-tf-module/blob/f0aecc6996effc288324204086453ee2e4671d33/variables.tf#L83-L93
I would like to see those moved down into `Optional parameters` at some point, but if you'd like to move it to a standalone PR I'm a-ok with that decision.

**Comment by jsf9k** [comment]
> I don't want to derail this lineage update so I'm approving because you've fixed my requests. However, I noticed that there are three variables with defaults in the `Required parameters` section:
> https://github.com/cisagov/openvpn-server-tf-module/blob/f0aecc6996effc288324204086453ee2e4671d33/variables.tf#L12-L16
> 
> 
> https://github.com/cisagov/openvpn-server-tf-module/blob/f0aecc6996effc288324204086453ee2e4671d33/variables.tf#L83-L93
> 
> 
> I would like to see those moved down into `Optional parameters` at some point, but if you'd like to move it to a standalone PR I'm a-ok with that decision.

See commit fba258e.

File: variables.tf [hunk 6]
```diff
@@ -167,18 +161,18 @@ variable "ssm_tlscrypt_key" {
   default     = "/openvpn/server/tlscrypt.key"
 }
 
+variable "ssm_read_role_accounts_allowed" {
+  type        = list(string)
+  description = "List of accounts allowed to access the role that can read SSM keys."
```
Comment (mcdonnnj) [inline variables.tf]: Just to keep this consistent with the other descriptions.

```suggestion
  description = "A list of accounts allowed to access the role that can read SSM keys."
```

**Review by mcdonnnj — APPROVED** [review]
...
image

File: variables.tf (hunk 6 above)
Comment (jsf9k) [reply variables.tf]: See commit d4302bc.

**Review by mcdonnnj — APPROVED** [review]
Thanks for your patience! 🚀✨

**Pull request merged**